pkg/specgen/generate/security.go

```diff
@@ -131,19 +131,16 @@ func securityConfigureGenerator(s *specgen.SpecGenerator, g *generate.Generator,
 	}
 
 	configSpec := g.Config
+	configSpec.Process.Capabilities.Ambient = []string{}
 	configSpec.Process.Capabilities.Bounding = caplist
+	configSpec.Process.Capabilities.Inheritable = caplist
 
 	if s.User == "" || s.User == "root" || s.User == "0" {
 		configSpec.Process.Capabilities.Effective = caplist
 		configSpec.Process.Capabilities.Permitted = caplist
-		configSpec.Process.Capabilities.Inheritable = caplist
 	} else {
-		userCaps, err := capabilities.NormalizeCapabilities(s.CapAdd)
-		if err != nil {
-			return errors.Wrapf(err, "capabilities requested by user are not valid: %q", strings.Join(s.CapAdd, ","))
-		}
-		configSpec.Process.Capabilities.Effective = userCaps
-		configSpec.Process.Capabilities.Permitted = userCaps
+		configSpec.Process.Capabilities.Effective = []string{}
```

Comment:
This is a breaking change.
In podman 1.9 we ran users with the permitted and effective capability when doing a --cap-add
This allows for containers like

$ podman run --cap-add net-bind-host --user 1234 fedora httpd

To work without ever needing a setuid application.

Reply:
I wasn't aware of such difference with Docker.

In this case, should we use your PR instead?

Reply:
Lets just restore this section:
```
} else {
		userCaps, err := capabilities.NormalizeCapabilities(s.CapAdd)
		if err != nil {
			return errors.Wrapf(err, "capabilities requested by user are not valid: %q", strings.Join(s.CapAdd, ","))
		}
		configSpec.Process.Capabilities.Effective = userCaps
		configSpec.Process.Capabilities.Permitted = userCaps
```
And go forward with this PR.

Reply:
fixed